### PR TITLE
Add isDestroyed check to instanceKeeperDispatcher

### LINKED ISSUE #69: InstanceKeeper should throw an error when used after destroy

### PR DESCRIPTION
**Pull request by VladislavSumin**
Closes #69

### CODE REVIEW

File: instance-keeper/src/commonMain/kotlin/com/arkivanov/essenty/instancekeeper/DefaultInstanceKeeperDispatcher.kt [hunk 1]
```diff
@@ -22,7 +27,12 @@ internal class DefaultInstanceKeeperDispatcher : InstanceKeeperDispatcher {
     override fun remove(key: Any): Instance? = map.remove(key)
```
Comment (arkivanov) [inline instance-keeper/src/commonMain/kotlin/com/arkivanov/essenty/instancekeeper/DefaultInstanceKeeperDispatcher.kt:27]: Would it be possible to also add the check for the `remove` method?

File: instance-keeper/src/commonMain/kotlin/com/arkivanov/essenty/instancekeeper/DefaultInstanceKeeperDispatcher.kt [hunk 2]
```diff
@@ -22,7 +27,12 @@ internal class DefaultInstanceKeeperDispatcher : InstanceKeeperDispatcher {
...
 
     override fun destroy() {
+        isDestroyed = true
```
Comment (arkivanov) [inline instance-keeper/src/commonMain/kotlin/com/arkivanov/essenty/instancekeeper/DefaultInstanceKeeperDispatcher.kt:30]: Would be also nice to call `checkIsNotDestroyed` before settings the flag.

**Review by arkivanov — CHANGES_REQUESTED** [review]

File: instance-keeper/src/commonMain/kotlin/com/arkivanov/essenty/instancekeeper/DefaultInstanceKeeperDispatcher.kt (hunk 2 above)
Comment (VladislavSumin) [reply instance-keeper/src/commonMain/kotlin/com/arkivanov/essenty/instancekeeper/DefaultInstanceKeeperDispatcher.kt:34]: done

**Review by arkivanov — APPROVED** [review]

**Comment by arkivanov** [comment]
Thanks for your contribution!

**Pull request merged**